src/main/java/org/jose/FibonacciBottomUp.java

```diff
@@ -1,32 +1,47 @@
 package org.jose;
 
 /**
- * Entry point.
+ * Esta clase define objetos de la clase Fibonacci BottomUp que se usan para
+ * calcular la suma de un número por este método.
  */
 public class FibonacciBottomUp implements FibonacciConMetodo {
     /**
-     * Entry point.
-     *
-     * @param n numero a calcular
-     * @return devuelve el resultado del calculo
+     * Campo de la clase.
      */
-    public int calcular(final int n) {
+    private int number;
+    /**
+     * Campo de la clase.
+     */
+    private ResultListener resultListener;
+    /**
+     * Constructor para Fibonacci BottomUp.
+     * @param numberToCalculate número a calcular.
+     * @param resultsListener .
+     */
+    public FibonacciBottomUp(final int numberToCalculate,
+                             final  ResultListener resultsListener) {
+        this.number = numberToCalculate;
+        this.resultListener = resultsListener;
+    }
+
+    /**
+     * Implementación método run, para el cálculo del número mediante el método
+     * BottomUp.
+     */
+    public void run() {
         int resultado = 1;
         int nmenos1 = 1;
         int nmenos2 = 1;
-        if (n == 1 || n == 0) {
-            return 1;
-        } else {
-            if (n > 0) {
-                for (int i = 2; i <= n; i++) {
-                    resultado = nmenos1 + nmenos2;
-                    nmenos2 = nmenos1;
-                    nmenos1 = resultado;
-                }
-                return resultado;
-            } else {
-                throw new RuntimeException("Error de calculo con: " + n);
+        if (number < 2) {
+            resultListener.onResult(1);
+        }
+        if (number > 0) {
+            for (int i = 2; i <= number; i++) {
```

Comment:
no te hace falta `if (number > 0)` porque en el `for` tu condicion es `i <= number`. si `number` es 0 o menor el for nunca va a empezar puesto quue el valor inicial de `i` es 2 y 2 es mayor que 0

Reply:
parate a leer el codigo para entender lo que digo y deja de cambiar cosas al tuntun a ver si aciertas